src/validation.cpp

```diff
@@ -3462,6 +3488,21 @@ bool CChainState::AcceptBlock(const std::shared_ptr<const CBlock>& pblock, CVali
         return error("%s: %s", __func__, FormatStateMessage(state));
     }
 
+    if (chainActive.Tip() == pindex->pprev) {
+        if (!AcceptStake(*pindex, block, state, *pcoinsTip)) {
+            return false;
+        }
+
+        // Header is valid, merkle tree and segwit merkle tree are good, stake is good...RELAY NOW
+        // (but if it does not build on our best tip, let the SendMessages loop relay it)
+        if (!IsInitialBlockDownload()) {
+            GetMainSignals().NewPoSValidBlock(pindex, pblock);
+        }
+    }
+    else {
+        // UNIT-E TODO:
```

Comment:
Nit: it would be nice to have slightly more detailed TODO here. It might not be obvious what can be done here.